src/files/getFile.api.js

```diff
@@ -1,62 +1,78 @@
 const path = require("path");
 const fs = require("fs-extra");
 
-const config = require("../config");
+const { logReceiver } = require("../logs/log-events");
 const buildApiHandler = require("../api-utils/build-api-handler");
 const userResolver = require("../middlewares/user-resolver");
 const {
   createParamValidator,
   REQ_COMPONENT,
 } = require("../middlewares/params-validator");
-
+const fileUtils = require("./file.utils")
 const filesService = require("./files.service");
-const { decipher } = require("../middlewares/file-decryption");
+const config = require("../config");
 
 async function controller(req, res) {
-  const { id } = req.query;
+  const { id } = req.params;
   const { user } = req.body;
 
   const existingFile = await filesService.getFile(id, user.username);
 
   if (!existingFile) {
+    logReceiver.emit(config.EVENT_NAME_LOG_COLLECTION, {
+      id,
+      username: user.username,
+      resMessage: `No file found for the id - ${id}`,
+    });
     res.json({
       message: `No file found for the id - ${id}`,
     });
     return;
   }
 
-  const READ_FILE_PATH = path.join(config.FILE_WRITE_DIRECTORY, id);
+  const destPath = path.join(config.DEST_DIR, id);
 
-  res.setHeader(
-    "Content-Type",
-    `${existingFile.mimeType}`
-  );
+  res.setHeader("Content-Type", `${existingFile.mimeType}`);
   res.setHeader(
     "Content-Disposition",
     `attachment; filename=${existingFile.fileName}`
   );
 
-  const fsReadStream = fs.createReadStream(READ_FILE_PATH);
+  const fsReadStream = fs.createReadStream(destPath);
 
-  fsReadStream.pipe(decipher).pipe(res);
+  fsReadStream.pipe(fileUtils.decipher).pipe(res);
```

Comment:
ciphering and deciphering functionalities should be moved into the service method itself